spec/models/miq_request_workflow_spec.rb

```diff
@@ -299,4 +302,57 @@
       expect(request.options[:owner_group]).to eq(owner.current_group.description)
     end
   end
+
+  context "#respool_to_folder" do
+    before(:each) do
+      resource_pool.ext_management_system = ems
+      ems_folder.ext_management_system = ems
+      attrs = ems_folder.attributes.merge(:object => ems_folder)
+      xml_hash = XmlHash::Element.new('EmsFolder', attrs)
+      hash = {"ResourcePool_#{resource_pool.id}" => xml_hash}
+      workflow.instance_variable_set("@ems_xml_nodes", hash)
+    end
+
+    it "returns nil if :respool is nil" do
+      respool = nil
+      src = {}.tap do |s|
+        s[:respool] = respool
+      end
```

Comment:
@syncrou As much as I like `tap` is this necessary?  It could be be reduced to a single line: `src = {:respool => respool}`.  I think it is more readable as well.  There are several occurrences like this in the new tests added.